internal/util/connection.go

```diff
@@ -56,7 +68,10 @@ func WriteToNetworkConnection(connection net.Conn, data []byte) error {
 
 	responseBuffer.Write(data)
 
-	_, err = connection.Write(responseBuffer.Bytes())
+	log.Trace("Writing the following bytes to a connection...", log.NewAttr("expected bytes", size))
+
+	numBytesWritten, err := connection.Write(responseBuffer.Bytes())
+	log.Trace("Wrote the following bytes to a connection.", log.NewAttr("actual bytes", numBytesWritten))
```

Comment:
In non-debugging, we generally don't want to get between a statement and it's error check.